scripts/build.sh

```diff
@@ -0,0 +1,239 @@
+            r) args["runtime-image"]="$OPTARG" ;;
+            -)
+                case "${OPTARG}" in
+                    help)
+                        usage; exit ;;
+                    deploy)
+                        args["deploy"]="YES" ;;
+                    no-persistence)
+                        args["no-persistence"]="YES" ;;
+                    image=*)
+                        assert_optarg_not_empty "$OPTARG" || exit $?
+                        args["image"]="${OPTARG#*=}"
+                    ;;
+                    namepsace=*)
+                        assert_optarg_not_empty "$OPTARG" || exit $?
+                        args["namepsace"]="${OPTARG#*=}"
+                    ;;
+                    manifests-directory=*)
+                        assert_optarg_not_empty "$OPTARG" || exit $?
+                        args["manifests-directory"]="$(realpath "${OPTARG#*=}" 2>/dev/null || echo "$PWD/${OPTARG#*=}")"
+                    ;;
+                    workflow-directory=*)
+                        assert_optarg_not_empty "$OPTARG" || exit $?
+                        args["workflow-directory"]="$(realpath "${OPTARG#*=}")" ;;
+                    builder-image=*)
+                        assert_optarg_not_empty "$OPTARG" || exit $?
+                        args["builder-image"]="${OPTARG#*=}"
+                    ;;
+                    runtime-image=*)
+                        assert_optarg_not_empty "$OPTARG" || exit $?
+                        args["runtime-image"]="${OPTARG#*=}"
+                    ;;
+                    *) log_error "Invalid option: --$OPTARG"; usage; exit 1 ;;
+                esac
+            ;;
+            \?) log_error "Invalid option: -$OPTARG"; usage; exit 2 ;;
+            :) log_error "Option -$OPTARG requires an argument."; usage; exit 3 ;;
+        esac
+    done
+
+    if [[ -z "${args["image"]:-}" ]]; then
+        log_error "Missing required flag: --image"
+        usage; exit 4
+    fi
+}
+
+function gen_manifests {
+    local res_dir_path="${args["workflow-directory"]}/src/main/resources"
+    local workflow_id
+    workflow_id="$(get_workflow_id "$res_dir_path")"
+
+    cd "$res_dir_path"
+    log_info "Switched directory: $res_dir_path"
+
+    local gen_manifest_args=(
+        -c="${args["manifests-directory"]}"
+        --profile='gitops'
+        --image="${args["image"]}"
+    )
+    if [[ -z "${args["namespace"]:-}" ]]; then
+        gen_manifest_args+=(--skip-namespace)
+    else
+        gen_manifest_args+=(--namespace="${args["namespace"]}")
+    fi
+    kn-workflow gen-manifest "${gen_manifest_args[@]}"        
+
+    cd "${args["workflow-directory"]}"
+    log_info "Switched directory: ${args["workflow-directory"]}"
+
+    # Find the sonataflow CR for the workflow
+    local sonataflow_cr
+    sonataflow_cr="$(findw "${args["manifests-directory"]}" -type f -name "*-sonataflow_${workflow_id}.yaml")"
+
+    if [[ -f secret.properties ]]; then
```

Comment:
should be `if [[ -f ${res_dir_path}/secret.properties ]]; then` otherwise the secret file will never be found and condition will never be executed